### PR TITLE
save relabelling mapping to csv

### PR DESCRIPTION
**Pull request by StevenShi-23**
Ref: #284 

- Summary: save `<old dataset pid, new pid>` to csv. This can be useful when users want to perform evaluation on training set using pre-trained model. In [current implementation](https://github.com/KaiyangZhou/deep-person-reid/blob/a9bcf6af2461d4bc58df5ad0359bb0dd17b5797d/torchreid/data/datasets/image/market1501.py#L56), train set will be relabelled for the continuity of label index. However, problems may arise because this mapping relationship is not visible to user, and is subject to different file indexing mechanism [(stackoverflow)](https://stackoverflow.com/questions/6773584/how-is-pythons-glob-glob-ordered). Therefore, training set may have different labels depending on machines, making it hard to reproduce the result. 
- Old behaviour: does not save remapping file.
- New behaviour: saves relabel mapping as csv to dataset directory. Eg: `market1501/train_relabel_18-13-31.csv`

### CODE REVIEW

File: torchreid/data/datasets/dataset.py [hunk 1]
```diff
@@ -293,6 +293,24 @@ def show_summary(self):
         )
         print('  ----------------------------------------')
 
+    def save_relabel_file(self, data_dir, relabel_map):
```
Comment (KaiyangZhou) [inline torchreid/data/datasets/dataset.py]: this function seems to be generic so we can move it to the `Dataset` class

File: torchreid/data/datasets/dataset.py [hunk 2]
```diff
@@ -293,6 +293,24 @@ def show_summary(self):
...
+        """
+        Save the mapping <pid_old, pid_new> after relabelling to csv.
+        :param data_dir: path to dataset that contains training set
+        :param relabel_map: a dict in the form of {pid_old:pid_new}
+        :return: None
+        """
+        import csv
+        from datetime import datetime
+        now = datetime.now()
+        relabel_record_path = osp.join(data_dir, "train_relabel"+now.strftime("_%H-%M-%S")+".csv")
```
Comment (KaiyangZhou) [inline torchreid/data/datasets/dataset.py]: will this create a new file everytime we run the program?
